src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -34,6 +34,12 @@ public override SizeRequest GetSizeRequest(double widthConstraint, double height
 
 		protected override Size MeasureOverride(double widthConstraint, double heightConstraint)
 		{
+			var margin = (this as IView)?.Margin ?? Thickness.Zero;
+
+			// Adjust the constraints to account for the margins
+			widthConstraint -= margin.HorizontalThickness;
+			heightConstraint -= margin.VerticalThickness;
+
 			var sizeWithoutMargins = LayoutManager.Measure(widthConstraint, heightConstraint);
```

Comment:
Might we rename this now that the margins are accounted for?

Reply:
It was inaccurate before. It's correct now. 

The old version wasn't removing the margins from the constraint size before measuring; now it's removing that space before measuring, so that measure _is_ the size without the margins included. It then adds the margins back in when setting the DesiredSize.